api/tech/oras/client.go

```diff
@@ -0,0 +1,201 @@
+package oras
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+
+	"github.com/containerd/containerd/errdefs"
+	ociv1 "github.com/opencontainers/image-spec/specs-go/v1"
+	oraserr "oras.land/oras-go/v2/errdef"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+type ClientOptions struct {
+	Client    *auth.Client
+	PlainHTTP bool
+}
+
+type Client struct {
+	client    *auth.Client
+	plainHTTP bool
+	ref       string
+	mu        sync.Mutex
+}
```

Comment:
On the risk of getting annoying now, I think what @jakobmoellerdev also mentioned in the discussion about this before is, that it would be better to actually use a `sync.RWMutex`. For the actual operations (`Fetch`, `List` and `Push`), a `RLock` would suffice then (I think?).

Reply:
Fine. :D